sbi/neural_nets/embedding_nets.py

```diff
@@ -268,17 +273,32 @@ def forward(self, x: Tensor) -> Tensor:
         """
         batch, permutation_dim, _ = x.shape
 
-        iid_embeddings = self.trial_net(x.view(batch * permutation_dim, -1)).view(
-            batch, permutation_dim, -1
-        )
+        # if no NaNs for padding varying trial lengths we can batch the computation
+        if not torch.isnan(x).any():
```

Comment:
how will this interact with `exclude_invalid_x=False`?

Reply:
the whole approach works only by setting `exclude_invalid_x=False`. if there are no nans we perform the embedding as before, without the loop over the batch.